python/cudf/cudf/core/dataframe.py

```diff
@@ -200,34 +200,41 @@ def __init__(self, data=None, index=None, columns=None, dtype=None):
         """
         super().__init__()
 
+        if isinstance(columns, (Series, cudf.Index)):
+            columns = columns.to_pandas()
+
         if isinstance(data, ColumnAccessor):
```

Comment:
This part is internal use only and we want it to be a fast path. Do we need to align to the `columns` here?

Reply:
I moved this logic to a before we branch out into any case to avoid code duplication per every case, but I think `columns` cudf object could be supplied in any of other valid cases too right?